### DISCUSSION TITLE
Publish container image to registry without draconian rate limits e.g. quay.io

### DISCUSSION THREAD

**Discussion by ibotty** [dani-garcia/vaultwarden#1782] [Ideas]
### Subject of the issue

It would be great to deploy to a container registry that does not rate limit image pulls (and checks for new image tags).   It's possible to apply to dockerhub to be exempt, but that comes with many strings attached (advertising from docker with your image, etc.).  I happen to like quay.io as a registry that does not limit pulls (except for abuse) and comes with a free account.

I get the following errors.
```
error: Import failed (InternalError): Internal error occurred: docker.io/vaultwarden/server:latest: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

### Deployment environment
* Install method:
Not yet.  Should be kubernetes.

### Steps to reproduce
`podman pull docker.io/vaultwarden/server:latest` too many times.

### Expected behaviour
`podman pull` succeeds.

### Actual behaviour
> docker.io/vaultwarden/server:latest: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit

**Comment by nalakawula** [comment]
I use cached docker image as workaround for docker hub rate limit, check this out:
[Pulling cached Docker Hub images](https://cloud.google.com/container-registry/docs/pulling-cached-images)

**Comment by RealOrangeOne** [comment]
GitHub has a registry which it'd be easy to push to as part of CI, and doesn't have rate limits on downloads.

**Comment by Felixkruemel** [comment]
> 
> 
> GitHub has a registry which it'd be easy to push to as part of CI, and doesn't have rate limits on downloads.

This would be amazing!

**Comment by BlackDex** [comment]
I'm going to move this to the discussions under `Ideas`, also to keep the issues for actual issues to the software.

**Comment by rguhr** [comment]
Image on quay.io would be great
